src/charms/request_extensions.cr

```diff
@@ -0,0 +1,9 @@
+class HTTP::Request
+  # This is an alternative to `remote_address`
+  # since that casts to `Socket::Address`, and not all
+  # subclasses have an `address` method to give you the value.
+  # ```
+  # request.remote_address.as?(Socket::IPAddress).try(&.address)
+  # ```
+  property remote_ip : String?
```

Comment:
Since this is a string, I wonder if there's a way this optional can be avoided? 

For the sake of argument, what if it defaulted to "0.0.0.0" if detection failed? What about an empty string?

Reply:
I think we can do that. Not sure if "0.0.0.0" is ok default... but at least an empty string just to avoid it being nilable.